src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RangeExtensions.cs

```diff
@@ -129,5 +135,15 @@ public static TextSpan AsTextSpan(this Range range, SourceText sourceText)
             var end = sourceText.Lines[(int)range.End.Line].Start + (int)range.End.Character;
             return new TextSpan(start, end - start);
         }
+
+        public static bool IsUndefined(this Range range)
+        {
+            if (range is null)
+            {
+                throw new ArgumentNullException(nameof(range));
+            }
+
+            return range == UndefinedRange;
```

Comment:
Should this be object.ReferenceEquals?

Reply:
I'm a bit hesitant to use ReferenceEquals for this as it's possible the undefined diagnostic may be coming from a different source (ex. deserialized stream) in which case we'd have two "undefined" ranges with different object references.